velox/functions/lib/Re2Functions.cpp

```diff
@@ -402,16 +424,44 @@ class LikeConstantPattern final : public VectorFunction {
     exec::DecodedArgs decodedArgs(rows, args, context);
     auto toSearch = decodedArgs.at(0);
     if (toSearch->isIdentityMapping()) {
-      auto rawStrings = toSearch->data<StringView>();
-      rows.applyToSelected([&](vector_size_t i) {
-        result.set(i, re2FullMatch(rawStrings[i], re_));
-      });
+      auto input = toSearch->data<StringView>();
+
+      if (fixedPattern_) {
+        rows.applyToSelected([&](vector_size_t i) {
+          result.set(i, likeutils::matchExactPattern(input[i], pattern_));
+        });
+      } else if (prefixPattern_) {
+        rows.applyToSelected([&](vector_size_t i) {
+          result.set(i, likeutils::matchPrefixPattern(input[i], pattern_));
+        });
+      } else if (suffixPattern_) {
+        rows.applyToSelected([&](vector_size_t i) {
+          result.set(i, likeutils::matchSuffixPattern(input[i], pattern_));
+        });
+      } else {
+        rows.applyToSelected([&](vector_size_t i) {
+          result.set(i, re2FullMatch(input[i], re_));
+        });
+      }
       return;
     }
 
     if (toSearch->isConstantMapping()) {
-      bool match = re2FullMatch(toSearch->valueAt<StringView>(0), re_);
-      rows.applyToSelected([&](vector_size_t i) { result.set(i, match); });
+      auto input = toSearch->valueAt<StringView>(0);
+
+      if (fixedPattern_) {
+        auto match = likeutils::matchExactPattern(input, pattern_);
+        rows.applyToSelected([&](vector_size_t i) { result.set(i, match); });
```

Comment:
This code repeats multiple time. It would be nice to compute the 'match' values first, then use this code to set the result values. Consider introducing a helper function that take an input and returns a boolean indicating whether it matches the pattern or not. This function can then be used here and above to reduce copy-paste.